### DISCUSSION TITLE
Is there a way to detect when link has been "fully" created?

### DISCUSSION THREAD

**Discussion by alexandernst** [clientIO/joint#2144] [Q&A] [answered]
### Introduction

I have a paper set with `linkPinning: false`. This lets me know when a link has been fully created (by checking the existence of a `target` cell). Or so I thought...

I have a use case in which I must render only the links that are embedded in a particular cell. If I start creating a new link inside that cell (by dragging the tip of a port of one embedded cell to the port of another embedded cell), there is a moment in which the link has both `source` and `target` ports, but hasn't been fully created because I haven't released the mouse (no `pointerup` even has been fired yet). And because no `pointerup` event has been fired yet, the link hasn't been `reparent()`-ed, which means that it hasn't been embedded into the cell.

Is there a way I can tell when a link has been "fully" created? "Fully" as in "is has a `source` and a `target` and the mouse has been released".

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment, marked as answer]
How about `paper.on('link:connect', () => { /* link was connected to a cell */ });`?

**Reply by alexandernst** [reply]
Yes, I was actually think about this and combining it with the other discussion I just opened about the `off()` method. I think this should work! Thank you!
